src/software/ai/hl/stp/play/free_kick/free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,317 @@
+                      receiver_positioning_tactics.end(),
+                      []() { return std::make_shared<MoveTactic>(); });
+    }
+
+    std::vector<Point> best_receiving_positions =
+        receiver_position_generator.getBestReceivingPositions(
+            *world, num_tactics, existing_receiver_positions, pass_origin_override);
+    // Note that getBestReceivingPositions may return fewer positions than requested
+    // if there are not enough robots, so we will need to check the size of the vector.
+    for (unsigned int i = 0;
+         i < receiver_positioning_tactics.size() && i < best_receiving_positions.size();
+         i++)
+    {
+        Angle receiver_orientation =
+            (world->ball().position() - best_receiving_positions[i]).orientation();
+        receiver_positioning_tactics[i]->updateControlParams(
+            best_receiving_positions[i], receiver_orientation, 0.0,
+            TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+            TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE);
+    }
+}
+
+void FreeKickPlayFSM::setReceiverAndDefenderTactics(
+    PriorityTacticVector &tactics_to_run, const Update &event, int ideal_num_receivers,
+    int num_tactics_already_assigned,
+    const std::vector<Point> &existing_receiver_positions,
+    const std::optional<Point> &pass_origin_override)
+{
+    int num_tactics_remaining = std::max(
+        0, static_cast<int>(event.common.num_tactics) - num_tactics_already_assigned);
+    int num_receivers = std::min(ideal_num_receivers, num_tactics_remaining);
+    int num_defenders = std::max(0, num_tactics_remaining - num_receivers);
+
+    if (num_receivers > 0)
+    {
+        updateReceiverPositioningTactics(event.common.world_ptr, num_receivers,
+                                         existing_receiver_positions,
+                                         pass_origin_override);
+        tactics_to_run[0].insert(tactics_to_run[0].end(),
+                                 receiver_positioning_tactics.begin(),
+                                 receiver_positioning_tactics.end());
+    }
+
+    defense_play->updateControlParams(TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT);
+
+    if (num_defenders > 0)
+    {
+        defense_play->updateTactics(PlayUpdate(
+            event.common.world_ptr, num_defenders,
+            [&tactics_to_run](PriorityTacticVector new_tactics) {
+                for (const auto &tactic_vector : new_tactics)
+                {
+                    tactics_to_run.push_back(tactic_vector);
+                }
+            },
+            event.common.inter_play_communication,
+            event.common.set_inter_play_communication_fun));
+    }
+}
+
+void FreeKickPlayFSM::updateAlignToBallTactic(const WorldPtr &world_ptr)
+{
+    // Face towards the center of the left segment of the enemy defense area,
+    // so we are prepared to take a shot on enemy net, or pass the ball near
+    // the enemy defense area.
```

Comment:
why specifically the negative x negative y corner and not just the enemy goal centre?

Reply:
This is similar to the one of the changes that Andrew had in his CreaseDefender PR. We're getting the center of the left segment of the defense area, and not the center of the defense area.

Reply:
why do we specifically choose that segment?